src/read/lazy.rs

```diff
@@ -0,0 +1,108 @@
+pub(crate) use imp::*;
+
+#[cfg(not(feature = "std"))]
+mod imp {
+    use alloc::sync::Arc;
+    use core::sync::atomic::{AtomicPtr, Ordering};
+    use core::{mem, ptr};
+
+    #[derive(Debug, Default)]
+    pub(crate) struct LazyArc<T> {
+        // Only written once with a value obtained from `Arc<T>::into_raw`.
+        value: AtomicPtr<T>,
```

Comment:
Maybe add a comment about how this is holding a ref count for us, so we can always safely load it and return a `Arc::clone` of it without fear of someone else racing in, decrementing its ref count to zero, and dropping it while we are still managing it / giving out clones?